base/tuple.jl

```diff
@@ -63,6 +63,15 @@ first(t::Tuple) = t[1]
 
 eltype(::Type{Tuple{}}) = Bottom
 eltype(::Type{<:Tuple{Vararg{E}}}) where {E} = E
+function eltype(t::Type{<:Tuple})
+    @_pure_meta
+    t´ = unwrap_unionall(t)
+    r = Union{}
+    for ti in t´.parameters
+        r = typejoin(r, unwrapva(ti))
```

Comment:
`rewrap_unionall` should be done at this point, since `typejoin` won't work on types with free variables (it calls `<:`).